--- src/xfeutils.cpporig	2010-04-29 20:31:23.000000000 +0200
+++ src/xfeutils.cpp	2010-04-29 20:32:44.000000000 +0200
@@ -512,7 +512,7 @@
 
 
 // Replacement of the stat function
-inline FXint statrep(const FXchar* filename, struct stat* buf)
+FXint statrep(const FXchar* filename, struct stat* buf)
 {
 #if defined(linux)
 
@@ -546,7 +546,7 @@
 
 
 // Replacement of the lstat function
-inline FXint lstatrep(const FXchar* filename, struct stat* buf)
+FXint lstatrep(const FXchar* filename, struct stat* buf)
 {
 #if defined(linux)
 
