--- ede-2.0-beta.orig/build/Program.jam	2009-10-08 10:44:35.000000000 -0300
+++ ede-2.0-beta/build/Program.jam	2009-10-11 09:31:50.000000000 -0300
@@ -76,8 +76,8 @@
 # and FLTKLIB. If [noinstall] is given, [target] will not be installed wit 'jam install'.
 rule EdeProgram
 {
-	if ! $(EDELIBINCLUDE) || ! $(EDELIBLIB) {
-		Echo "EDELIBINCLUDE or EDELIBLIB not defined; $(1) will not be built" ;
+	if ! $(EDELIBLIB) {
+		Echo "EDELIBLIB not defined; $(1) will not be built" ;
 		return ;
 	}
 
@@ -95,8 +95,8 @@
 # and FLTKLIB. If [noinstall] is given, [target] will not be installed wit 'jam install'.
 rule EdeProgramAsFltkBare
 {
-	if ! $(EDELIBINCLUDE) || ! $(EDELIBLIB) {
-		Echo "EDELIBINCLUDE or EDELIBLIB not defined; $(1) will not be built" ;
+	if ! $(EDELIBLIB) {
+		Echo "EDELIBLIB not defined; $(1) will not be built" ;
 		return ;
 	}
 
