--- ede-2.0-beta.orig/ede-bug-tools/ede-bug-report/xmlrpc-c/deps/util/include/stdargx.h	2009-10-08 10:44:45.000000000 -0300
+++ ede-2.0-beta/ede-bug-tools/ede-bug-report/xmlrpc-c/deps/util/include/stdargx.h	2009-10-10 12:23:27.000000000 -0300
@@ -60,7 +60,7 @@
     */
     memcpy(&argsxP->v, args, sizeof(argsxP->v));
 #else
-    argsxP->v = args;
+    va_copy(argsxP->v, args);
 #endif
 }
 
--- ede-2.0-beta.orig/ede-autostart/ede-autostart.cpp	2009-10-08 10:44:55.000000000 -0300
+++ ede-2.0-beta/ede-autostart/ede-autostart.cpp	2009-10-11 09:37:21.000000000 -0300
@@ -87,8 +87,8 @@
 static Fl_Check_Browser* cbrowser;
 static Fl_Pixmap         warnpix((const char**)warning_xpm);
 
-static char* get_basename(const char* path) {
-	char* p = strrchr(path, '/');
+static const char* get_basename(const char* path) {
+	const char* p = strrchr(path, '/');
 	if(p) 
 		return (p + 1);
 
--- ede-2.0-beta.orig/ede-desktop/Utils.cpp	2009-10-08 10:44:06.000000000 -0300
+++ ede-2.0-beta/ede-desktop/Utils.cpp	2009-10-11 09:48:07.000000000 -0300
@@ -305,11 +305,11 @@
 }
 
 char* get_basename(const char* path) {
-	char* p = strrchr(path, '/');
+	const char* p = strrchr(path, '/');
 	if(p)
-		return (p + 1);
+		return const_cast<char*>(p + 1);
 
-	return (char*)path;
+	return const_cast<char*>(path);
 }
 
 bool is_temp_filename(const char* path) {
--- ede-2.0-beta.orig/ede-launch/ede-launch.cpp	2009-10-08 10:44:09.000000000 -0300
+++ ede-2.0-beta/ede-launch/ede-launch.cpp	2009-10-11 09:53:11.000000000 -0300
@@ -67,7 +67,7 @@
 }
 
 static char* get_basename(const char* path) {
-	char *p = strrchr(path, '/');
+	char *p = const_cast<char*>(strrchr(path, '/'));
 	if(p)
 		return (p + 1);
 
