diff -u edelib-2.0-beta.orig/src/IconChooser.cpp edelib-2.0-beta/src/IconChooser.cpp
--- edelib-2.0-beta.orig/src/IconChooser.cpp	2009-10-08 10:43:55.000000000 -0300
+++ edelib-2.0-beta/src/IconChooser.cpp	2009-10-10 10:58:52.000000000 -0300
@@ -82,7 +82,7 @@
 	iconpth.assign(s);
 
 	/* get basename without extension */
-	ptr = strrchr(s.c_str(), E_DIR_SEPARATOR);
+	ptr = const_cast<char*>(strrchr(s.c_str(), E_DIR_SEPARATOR));
 	if(ptr) {
 		ptr += 1;
 		char* ptr2 = strrchr(ptr, '.');
diff -u edelib-2.0-beta.orig/src/String.cpp edelib-2.0-beta/src/String.cpp
--- edelib-2.0-beta.orig/src/String.cpp	2009-10-08 10:43:55.000000000 -0300
+++ edelib-2.0-beta/src/String.cpp	2009-10-10 10:31:05.000000000 -0300
@@ -262,7 +262,7 @@
 	if(offset >= length())
 		return npos;
 
-	char* p = strstr(data() + offset, str);
+	const char* p = strstr(data() + offset, str);
 	if(!p)
 		return npos;
 	else
