--- kdenlive-0.5.orig/kdenlive/docclipavfile.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/docclipavfile.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -18,7 +18,7 @@
 #include "docclipavfile.h"
 
 #include <iostream>
-#include <assert.h>
+#include <cassert>
 
 #include <qfileinfo.h>
 
--- kdenlive-0.5.orig/kdenlive/docclipproject.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/docclipproject.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -17,7 +17,7 @@
 
 #include "docclipproject.h"
 
-#include <assert.h>
+#include <cassert>
 #include <qptrvector.h>
 #include <qvaluevector.h>
 #include <qptrlist.h>
--- kdenlive-0.5.orig/kdenlive/doccliptextfile.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/doccliptextfile.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -19,7 +19,7 @@
 #include "doccliptextfile.h"
 
 #include <iostream>
-#include <assert.h>
+#include <cassert>
 
 #include <qfileinfo.h>
 
--- kdenlive-0.5.orig/kdenlive/docclipvirtual.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/docclipvirtual.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -19,7 +19,7 @@
 #include "docclipvirtual.h"
 
 #include <iostream>
-#include <assert.h>
+#include <cassert>
 
 #include <qfileinfo.h>
 
--- kdenlive-0.5.orig/kdenlive/docsubclip.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/docsubclip.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -22,7 +22,7 @@
 
 #include <iostream>
 #include <kdebug.h>
-#include <assert.h>
+#include <cassert>
 DocSubClip::DocSubClip(KdenliveDoc * doc, DocClipBase * clip):
 m_clip(clip)
 {
--- kdenlive-0.5.orig/kdenlive/dynamicToolTip.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/dynamicToolTip.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -17,7 +17,7 @@
 #include "dynamicToolTip.h"
 #include <qapplication.h>
 #include <qpainter.h>
-#include <stdlib.h>
+#include <cstdlib>
 
 #include <kdebug.h>
 
--- kdenlive-0.5.orig/kdenlive/effectdesc.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/effectdesc.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -15,7 +15,7 @@
  *                                                                         *
  ***************************************************************************/
 
-#include <assert.h>
+#include <cassert>
 #include "kdebug.h"
 
 #include "effectdesc.h"
--- kdenlive-0.5.orig/kdenlive/effectparamdialog.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/effectparamdialog.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -27,7 +27,7 @@
 #include <klocale.h>
 #include <kpushbutton.h>
 
-#include <assert.h>
+#include <cassert>
 
 #include <kfixedruler.h>
 #include <kmmtimeline.h>
--- kdenlive-0.5.orig/kdenlive/kdenlive.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/kdenlive.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -17,6 +17,8 @@
 
 #define _ISOC99_SOURCE
 
+#include <cstdlib>
+
 // include files for QT
 #include <qdir.h>
 #include <qprinter.h>
--- kdenlive-0.5.orig/kdenlive/kmmrulerpanel.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/kmmrulerpanel.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -26,8 +26,8 @@
 #include <kdebug.h>
 
 
-#include <stdlib.h>
-#include <math.h>
+#include <cstdlib>
+#include <cmath>
 
 namespace Gui {
 
--- kdenlive-0.5.orig/kdenlive/kmmscreen.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/kmmscreen.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -16,7 +16,7 @@
  ***************************************************************************/
 
 #include "kmmscreen.h"
-#include <string.h>
+#include <cstring>
 #include <iostream>
 #include <kdebug.h>
 #include <klocale.h>
--- kdenlive-0.5.orig/kdenlive/krender.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/krender.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -27,6 +27,7 @@
 }
 
 #include <iostream>
+#include <cstdlib>
 
 // ffmpeg Header files
 
--- kdenlive-0.5.orig/kdenlive/kruler.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/kruler.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -24,6 +24,7 @@
 
 #include <iostream>
 #include <cmath>
+#include <cstdlib>
 
 #include <kdebug.h>
 #include <kstyle.h>
--- kdenlive-0.5.orig/kdenlive/krulertimemodel.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/krulertimemodel.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -16,7 +16,7 @@
  ***************************************************************************/
 
 #include <kdebug.h>
-#include <math.h>
+#include <cmath>
 
 #include "krulertimemodel.h"
 #include "kdenlivesettings.h"
--- kdenlive-0.5.orig/kdenlive/ktimeline.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/ktimeline.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -17,9 +17,9 @@
 
 #include <cmath>
 #include <iostream>
-#include <stdlib.h>
+#include <cstdlib>
 #include <iostream>
-#include <assert.h>
+#include <cassert>
 
 #include <qscrollbar.h>
 #include <qscrollview.h>
--- kdenlive-0.5.orig/kdenlive/ktrackpanel.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/ktrackpanel.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -15,7 +15,7 @@
  *                                                                         *
  ***************************************************************************/
 #include "ktrackpanel.h"
-#include <assert.h>
+#include <cassert>
 
 #include "kplacer.h"
 #include "trackviewdecorator.h"
--- kdenlive-0.5.orig/kdenlive/ktrackview.cpp	2007-08-12 16:16:52.000000000 +0800
+++ kdenlive-0.5/kdenlive/ktrackview.cpp	2008-05-07 11:01:59.000000000 +0800
@@ -16,7 +16,8 @@
  ***************************************************************************/
 
 #include <iostream>
-#include <math.h>
+#include <cmath>
+#include <cstdlib>
 
 #include <qpainter.h>
 #include <qcursor.h>

