--- configure	（版本 26219）
+++ configure	（工作副本）
@@ -5420,7 +5423,7 @@
 elif test "$_dvdread" = yes; then
   _def_dvdread='#define USE_DVDREAD 1'
   _largefiles=yes
-  _ld_extra="$_ld_extra -ldvdread"
+  _ld_extra="$_ld_extra -ldvdnav -ldvdread"
   _inputmodules="dvdread(external) $_inputmodules"
   _res_comment="external"
 else
@@ -7620,7 +7624,7 @@
   _largefiles=yes
   _def_dvdnav='#define USE_DVDNAV 1'
   _inc_extra="$_inc_extra `$_dvdnavconfig --cflags`"
-  _ld_extra="$_ld_extra `$_dvdnavconfig --minilibs`"
+  _ld_extra="$_ld_extra `$_dvdnavconfig --libs`"
   _inputmodules="dvdnav $_inputmodules"
 else
   _def_dvdnav='#undef USE_DVDNAV'
